packages/nameguard-react/src/components/ReportBadge/index.tsx

```diff
@@ -0,0 +1,85 @@
+import { type ConsolidatedNameGuardReport } from "@namehash/nameguard";
+import React from "react";
+import cc from "classcat";
+
+import { ReportIcon } from "../ReportIcon/index";
+import { RatingLoadingIcon, RatingIconSize } from "../..";
+import { UnknownReportShield } from "../UnknownReportShield/UnknownReportShield";
+
+interface ReportBadgeProps {
+  name: string;
+  onClick?: () => any;
+  data?: ConsolidatedNameGuardReport;
+  hadLoadingError?: boolean;
+}
+
+export function ReportBadge({
+  name,
+  data,
+  onClick,
+  hadLoadingError,
+}: ReportBadgeProps) {
+  const buttonClass =
+    "flex-shrink-0 appearance-none bg-white transition-colors hover:bg-gray-50 border border-gray-200 rounded-md px-2.5 py-1.5 space-x-1.5 inline-flex items-center";
+  const buttonAndCursorClass = cc([buttonClass, "cursor-pointer"]);
+
+  const onClickHandler = () => {
+    if (onClick) onClick();
+    else {
+      window.location.href = `https://nameguard.io/inspect/${encodeURIComponent(
+        data.name,
+      )}`;
+    }
+  };
+
+  if (hadLoadingError)
+    return (
+      <button className={buttonAndCursorClass} onClick={onClickHandler}>
+        <span className="text-black text-sm leading-5 ens-webfont">{name}</span>
+        <UnknownReportShield size="micro" className={"cursor-pointer"}>
+          <div className="text-sm text-white">
+            <button
+              className="appearance-none underline font-medium"
+              onClick={onClickHandler}
+            >
+              Inspect name for details
+            </button>
+          </div>
+        </UnknownReportShield>
+      </button>
+    );
+
+  if (!data)
+    return (
+      <button className={buttonAndCursorClass} onClick={onClickHandler}>
+        <span className="text-black text-sm leading-5 ens-webfont">{name}</span>
+        <RatingLoadingIcon
+          size={RatingIconSize.small}
+          className={"cursor-pointer"}
+        />
+      </button>
+    );
+
+  return (
+    <button className={buttonAndCursorClass} onClick={onClickHandler}>
+      <span className="text-black text-sm leading-5 ens-webfont">
+        {data.beautiful_name || data.name}
```

Comment:
Ah, there's a problem here.

It's possible that data.name is different than data.beautiful_name. This is the difference between the normalized version of a name and the beautiful version of a name. For example, consider the name: "ξsk3nder.eth". The current logic will have the name being displayed suddenly change once the data is loaded.

The fix to this is to stop passing `name` as a string and instead pass it as an `ENSName` instead as defined in `ens-utils`. Then this UI component will can always show `name.displayName` where we show the name to the user. For cases in these UI components where we generate a link, we should instead be using `name.name`.

You can use the `buildENSName` function in `ens-utils` to convert from name as a string to name as an `ENSName`.

Please ask me if any questions.